update actions versions

stojanovskis1 opened the pull request:
Update github actions, ticket #88

KirilPoposki98 reviewed (approved):
Everything looks good, proceeding with testing

stojanovskis1 commented:
Added new commits for changing set-output deprecated command to $GITHUB_OUTPUT, updated codeql-action to v3.
please re-review @KirilPoposki98 @Filip3mac
